wdl2cwl/tests/wdl_files/blast.wdl

```diff
@@ -0,0 +1,325 @@
+version 1.0
+
+workflow blast {
+  input {
+    String? blast_docker_override
+    String  blast_docker= select_first([blast_docker_override,"swr.cn-south-1.myhuaweicloud.com/cngbdb/blast:1.2"])
+    File    queryfa
+    String  fname       = '/sfs/blastdb/2019_ncov/nucl/v6/ncov'
+    String  method      = 'blastn'
+    Int     outfmt      = 7
+    Float   evalue      = 10
+    String  Outfile     = basename(queryfa)+'.blast_result.txt'
+    Int     threads     = 8
+  }
+  if (method == 'blastp') {
+    call runblastp{
+      input:
+        Fname      = fname,
+        Queryfa    = queryfa,
+        docker     = blast_docker,
+        outfmt     = outfmt,
+        evalue     = evalue,
+        Outfile    = Outfile,
+        threads    = threads
+    }
+  }
+  if ( method == 'blastn'){
+    call runblastn{
+      input:
+        Fname      = fname,
+        Queryfa    = queryfa,
+        docker     = blast_docker,
+        outfmt     = outfmt,
+        evalue     = evalue,
+        Outfile    = Outfile,
+        threads    = threads
+    }
+  }
+  if ( method == 'blastx'){
+    call runblastx{
+      input:
+        Fname      = fname,
+        Queryfa    = queryfa,
+        docker     = blast_docker,
+        outfmt     = outfmt,
+        evalue     = evalue,
+        Outfile    = Outfile,
+        threads    = threads
+    }
+  }
+  if ( method == 'queryfa'){
+    call runtblastn{
+      input:
+        Fname      = fname,
+        Queryfa    = queryfa,
+        docker     = blast_docker,
+        outfmt     = outfmt,
+        evalue     = evalue,
+        Outfile    = Outfile,
+        threads    = threads
+    }
+  }
+  if ( method == 'tblastx'){
+    call runtblastx{
+      input:
+        Fname      = fname,
+        Queryfa    = queryfa,
+        docker     = blast_docker,
+        outfmt     = outfmt,
+        evalue     = evalue,
+        Outfile    = Outfile,
+        threads    = threads
+    }
+  }
+  output {
+    File fina_output =select_first([runtblastx.out,runblastp.out,runblastn.out,runblastx.out,runtblastn.out])
```

Comment:
And it wasn't even a complete (mis)translation 😅

Reply:
It looks like I cannot call `get_expr` for all expressions, as it messed up the `outputSoruce` values in other files, breaking some tests.

I changed it to handle `Expr.Get` as before, but for others to call `get_expr`. Still not looking 100%, since the expression appears to be partially evaluated? :confused: Pushed the latest CWL file and changes.